### PR TITLE
test(osio): Correctly clean OsioEndpointIT data

### LINKED ISSUE #287: Check that OsioEndpointIT cleans everything correctly
- I think osio codebase are not cleaned 
- @gastaldi said that maybe some GH repo too

### PR DESCRIPTION
**Pull request by ia3andy**
### Why

Codebases were not deleted after test execution.

### Description

Create and delete the space itself in order to clean all data after the test.

Closes #287

### Checklist

- [x] I followed the contribution [guidelines](https://raw.githubusercontent.com/fabric8-launcher/launcher-backend/master/CONTRIBUTING.md).
- [x] I created an [issue](https://github.com/fabric8-launcher/launcher-backend/issues/new) and used it in the commit message.
- [x] I have built the project locally prior to submission with `mvn clean install`.
- [x] I kept a clean commit log (no unnecessary commits, no merge commits).
- [x] I am happy with my contribution.

### CODE REVIEW

**Comment by gastaldi** [comment]
[test]

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Space findSpaceById(final String spaceId) {
     public Space findSpaceByName(final String tenantName, final String spaceName) {
         final Request request = newAuthorizedRequestBuilder("/api/namedspaces/" + tenantName + "/" + spaceName).build();
         return executeAndParseJson(request, OsioWitClient::readSpace)
-                .orElseThrow(() -> new IllegalArgumentException("Space not found for tenant:" + tenantName + " with name " + spaceName));
+                .orElseThrow(() -> new IllegalStateException("Space not found for tenant:" + tenantName + " with name " + spaceName));
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: When this exception is thrown, the argument is OK but the "space" itself is not found for this argument. IMO using IllegalArgumentException is wrong in that kind of case.. IllegalStateException too, it should some kind of ItemNotFoundException if we were really strict but now that you say that, I think we should return an Optional directly. WDYT?

[the remark this answers is not in the document]
Comment (gastaldi) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: Ah yes, returning an Optional would make more sense I guess.

File: addons/osio-addon/src/test/java/io/fabric8/launcher/osio/client/OsioWitClientTest.java [hunk 2]
```diff
@@ -52,26 +66,36 @@ public void shouldGetTenantCorrectly() {
...
     @Test
     public void shouldCreateCodeBaseCorrectly() {
-        final Space space = getOsioWitClient().findSpaceById("89809077-f220-40a4-897b-cc1b99ff95ca");
+        final Space space = getOsioWitClient().findSpaceById(defaultSpace.getId());
         getOsioWitClient().createCodeBase(space.getId(), "stack", URI.create("https://github.com/ia3andy/hoob.git"));
     }
 
+    @Test
+    public void shouldCreateAndDeleteSpaceCorrectly() {
+        final Space space = getOsioWitClient().createSpace("test-wit-client-create");
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply addons/osio-addon/src/test/java/io/fabric8/launcher/osio/client/OsioWitClientTest.java]: OsioWitClient contains some code, the virtualization let us unit test this code. It's just like using a mock, but easier to record.

[the remark this answers is not in the document]
Comment (gastaldi) [reply addons/osio-addon/src/test/java/io/fabric8/launcher/osio/client/OsioWitClientTest.java]: I guess this is the same discussion we had in the past about adopting Hoverfly. I don't think we have much value in testing a mocked service, but maybe it's just me

**Pull request merged**